cmd/admin-revoker/main.go

```diff
@@ -235,6 +227,26 @@ func main() {
 				}
 			},
 		},
+		{
+			Name:  "auth-revoke",
+			Usage: "Revoke all pending/valid authorizations for a domain",
+			Action: func(c *cli.Context) {
+				domain := c.Args().First()
+				_, _, _, sac := setupContext(c)
+				auths, err := sac.GetAuthorizationsByDomain(core.AcmeIdentifier{Type: core.IdentifierDNS, Value: domain})
+				if err != nil {
+					cmd.FailOnError(err, "Failed to retrieve authoriations")
+				}
+				for _, a := range auths {
+					if a.Status != core.StatusInvalid && a.Status != core.StatusRevoked {
+						err = sac.RevokeAuthorization(a)
+						if err != nil {
+							cmd.FailOnError(err, "Failed to revoke authorization")
```

Comment:
Give context on which authorization failed. There might be Boogs for specific domains or authorizations.